uefi-raw/src/protocol/network/tcp4.rs

```diff
@@ -0,0 +1,503 @@
+
+    /// Add or delete routing entries.
+    ///
+    /// The most specific route is selected by comparing the
+    /// `subnet_address` with the destination IP address
+    /// arithmetically ANDed with the `subnet_mask`.
+    ///
+    /// The default route is added with both `subnet_address` and
+    /// `subnet_mask` set to `0.0.0.0`. The default route matches all
+    /// destination IP addresses if there is no more specific route.
+    ///
+    /// A direct route is added with `gateway_address` set to
+    /// `0.0.0.0`. Packets are sent to the destination host if its
+    /// address can be found in the Address Resolution Protocol (ARP)
+    /// cache or it is on the local subnet. If the instance is
+    /// configured to use default address, a direct route to the local
+    /// network will be added automatically.
+    ///
+    /// Each TCP instance has its own independent routing table. An
+    /// instance that uses the default IP address will have a copy of
+    /// the
+    /// [`Ipv4Config2Protocol`][super::ip4_config2::Ip4Config2Protocol]'s
+    /// routing table. The copy will be updated automatically whenever
+    /// the IP driver reconfigures its instance. As a result, the
+    /// previous modification to the instance's local copy will be
+    /// lost.
+    ///
+    /// The priority of checking the route table is specific to the IP
+    /// implementation, and every IP implementation must comply with
+    /// RFC 1122.
+    ///
+    /// Note: There is no way to set up routes to other network
+    /// interface cards (NICs) because each NIC has its own
+    /// independent network stack that shares information only through
+    /// EFI TCPv4 variable.
+    pub routes: unsafe extern "efiapi" fn(
+        this: *mut Self,
+        delete_route: Boolean,
+        subnet_address: *const Ipv4Address,
+        subnet_mask: *const Ipv4Address,
+        gateway_address: *const Ipv4Address,
+    ) -> Status,
+
+    /// Initiate a nonblocking TCP connection request for an active
+    /// TCP instance.
+    ///
+    /// `connect` initiates an active open to the remote peer
+    /// configured in the current TCP instance if it is configured as
+    /// active. If the connection succeeds or fails due to any error,
+    /// the [`connection_token.event`][Tcp4CompletionToken::event]
+    /// will be signaled and `connection_token.status` will be updated
+    /// accordingly. This function can only be called for the TCP
+    /// instance in the [`Tcp4ConnectionState::CLOSED`] state. The
+    /// instance will transition to [`Tcp4ConnectionState::SYN_SENT`]
+    /// if the function returns [`Status::SUCCESS`]. If the TCP
+    /// three-way handshake succeeds, its state will become
+    /// [`Tcp4ConnectionState::ESTABLISHED`], otherwise, the state
+    /// will return to [`Tcp4ConnectionState::CLOSED`].
+    pub connect: unsafe extern "efiapi" fn(
+        this: *mut Self,
+        connection_token: *mut Tcp4CompletionToken,
```

Comment:
once last thing. Please add `pub type Tcp4ConnectionToken = Tcp4CompletionToken` and use the type alias here.

<img width="1146" height="314" alt="Image" src="https://github.com/user-attachments/assets/22a84b63-d384-4a4e-b3cf-de1532a8ca1b" />

Reply:
Fine with me